### PR TITLE
Update askforhelp.rst

### PR DESCRIPTION
**Pull request by tkimnguyen**
wording tweaks, genericize chat, include Gitter, update links, add tips

Fixes: #

Improves:

Changes proposed in this pull request:

-

-

-

### CODE REVIEW

File: askforhelp.rst [hunk 1]
```diff
@@ -63,17 +63,16 @@ The big picture - An opening sentence should state the general problem that you
...
-**Steps to reproduce the issue** - Give information about your ideas of how this error appeared, what caused it or anything that could lead to reproducing the error on another computer. The expected result .
+**Steps to reproduce the issue** - Give information about your ideas of how this error appeared, what caused it or anything that could lead to reproducing the error on another computer, including your buildout.cfg, versions.cfg, version numbers of installed add-ons, detailed command lines, the complete error message stack. Mention your expected result.
 
-Asking help in real time chat (IRC)
------------------------------------
+Asking for help in online chat
+------------------------------
 
-Plone community real time chat is at the  `#plone <irc://irc.freenode.net/#plone>`_ IRC channel on the freenode IRC network.
+The Plone community uses `online chat <https://plone.org/support/chat>`_, specifically `Gitter <https://gitter.im/plone/public>`_ and traditional Plone IRC chat at the  `#plone <irc://irc.freenode.net/#plone>`_ IRC channel on the freenode IRC network.
```
Comment (svx) [inline askforhelp.rst:72]: IMHO if we want to mention Gitter, there should be more information about it, like that you will need a GitHub or Twitter account to sign in.

Comment (svx) [inline askforhelp.rst:72]: IMHO if we want to mention Gitter, there should be more information about it, like that you will need a GitHub or Twitter account to sign in.

Comment (svx) [reply askforhelp.rst:72]: Next question why Gitter, there are also people on Slack so what is the idea behind adding Gitter but not Slack ?

IMHO I would be really careful with adding more and more options here, we should make it 'easy' for user and thsi means add the option(s) which the most people are using.
Otherwise you could end up in one channel which is not active at all.

So yeah I am not against it, but like I said we should be careful with splitting our own resources here.
It should not about what you or me personally like, it should be about which one is 'easy' and used by the most.

**Comment by stevepiercy** [comment]
FWIW, Pyramid and other projects under the Pylons Project went through the same thing. We debated over issues such as open vs. closed source, account required or not, spreading ourselves thinly. Ultimately we opted for email via Google Groups and IRC as our two official support channels, both of which are open and archived.

File: askforhelp.rst (hunk 1 above)
Comment (tkimnguyen) [reply askforhelp.rst:72]: It does say that in the room's welcome message. The canonical info about Gitter is at plone.org/support/chat so I didn't just want to repeat the info and risk having it get outdated eventually.

Comment (tkimnguyen) [reply askforhelp.rst:72]: I agree and understand. We went through this discussion and ended up deciding to leave Slack fans on their own and able to use Slack (they continue to do so), but for official chat we leave IRC for fans of that and we introduce Gitter because it's friendly to open source projects, doesn't have the problems Slack has caused for open source projects, and it authenticates with GitHub, which makes that match plone.org and our code repos and issue trackers. Also, Gitter's requirement for authentication lets us respond to questions & notify people after they'd normally have disconnected from IRC (something that drives me bananas not just personally but also from the community perception perspective).

In other words, we preserve choice and freedom but steer newcomers to Gitter if they don't like or find IRC unapproachable (which we've heard a lot of).

**Comment by tkimnguyen** [comment]
thx @stevepiercy - our equivalent to Google Groups is https://community.plone.org where we have search and archiving

File: askforhelp.rst (hunk 1 above)
Comment (svx) [reply askforhelp.rst:72]: Would you mind to point me to the place where "we" made the decision? It seems I missed that, I only can find postings on community where people discus that.

Besides from my own opinion why this is *really bad* I will try to explain it in a hopefully different way, sorry that is getting 'longish' now :).

Please try to think not as Kim the developer but in place of that as Kim the webmaster who is new to Plone and never used it before.
So you are a webmaster, you do not have programmer knowledge, you do not write code, your work environment is your browser and Libre Office.

You have a question about a setting in the CMF, you can't find the answer in the docs (bad bad) but hey there is this chat mentioned on docs.plone.org/askforhelp. Hopefully there people will able to answer me my question, yeah !
Your click on the link and you end up on the Gitter page, nice, but uh eh I can see there is a chat and I can see people talking but I can not ask a question, first I have to login ...

Ok lets click on sign in ... uh they are asking for a GitHub or Twitter login .. What is GitHub never heard about it, lest check oh for developer and nerds, ok do not need nor have that .... Twitter? Hm .. I will not use my personally account to login for work related stuff, and I am not allowed to use a company account ....

So hey, looks nice but I am not able to use that in anyway ...

Thanks for staying with me !

What I am trying to say, Gitter maybe handy for you and I can see the point of have all the nice GitHub hooks, but I doubt that is that useful for 'normal' people.
So IMHO you make it harder for people to ask for help. If you are a 'developer',a person who has already a GitHub account, yes it maybe helpful, great but for others not at all, you may even push them away, which is bad for Plone.

That is exactly the reason why we thought about this deep and long for docs.plone.org and went with a really 'simple' widget, which people can click and just can write us or attach pictures, no need for a login somewhere.
Again ... if you want to be serious about helping people you also should make it easy and in a way that it is working for them, this may involve slightly more work you you, but like I said it is not about you it is about user.
Going further this is also about marketing, if you make it hard for people you will loose them to other Solutions and you will get knows as hard to use and impossibly to get help.

If I went for all stuff what and how I would like to do docs.plone.org they would be completely different today, I did not do that because it would be 'harder' for people to contribute, so I did not do that, yet ......(you never know), hehe
I you feel that you get 'bananas' the way how IRC works and people disconnect, maybe the discussion should be more 'do we want a chat as support' at all ?

So in short I am serious concerned and against merging that in the current state. If this would get re-written in a way, where it is better and more explained that Gitter requires to login and also the reason why 'we' choose for that, I may fine, but sorry not how it is now.

@tkimnguyen please do not take it personally, this is **not meant** to insult you at all ! I am trying to make it better for less technical people because I am concerned that we 'forget' them again and I am concerned about the way Plone is moving .....

Comment (svx) [reply askforhelp.rst:72]: see my long answer above :)

**Comment by tkimnguyen** [comment]
@svx I don't really care to argue this further in here. We want to improve the current state of chat, we have done that, and now we need to update docs to reflect that. If you don't want to merge this then don't but be aware that the page as it stands has a bunch of other issues.

**Comment by svx** [comment]
@tkimnguyen your current PR is IMHO not improving the situation, that is what I am trying to say, currently you will leave the person who is looking for help with various chat possibilities [IRC and Gitter] without giving or telling a clear direction and because of this this is confusing which one should the user choose, like I tried to explain above, if you really like to add Gitter you should make sure to explain why this could be an advantage over IRC, like *if you already using GitHub* bla bla bla.
But there should be also explained that you need to login with GitHub/Twitter.

So maybe it would be better for all of us [community] to define first what kind of support/chat we want to provide to/for which audience, this is what I am still missing, we as community have currently, at least so it looks to me no idea about who/what/how ...

The point if using Gitter is an improvement is a completely different topic... I am still serious concerned about Gitter and the reasons why and how the decision was made ....

**Comment by tkimnguyen** [comment]
@svx I hope I've clarified it based on your suggestions. I appreciate and can act on specific suggestions.

**Comment by svx** [comment]
Needs to cherry-picked into 5.1

**Comment by svx** [comment]
Thank you !

**Comment by svx** [comment]
Merged into 5.1 !
